autofile/group.go

```diff
@@ -596,14 +596,12 @@ func (gr *GroupReader) Read(p []byte) (n int, err error) {
 		nn, err = gr.curReader.Read(p[n:])
 		n += nn
 		if err == io.EOF {
-			// Open the next file
-			if err1 := gr.openFile(gr.curIndex + 1); err1 != nil {
-				return n, err1
-			}
 			if n >= lenP {
 				return n, nil
-			} else {
-				continue
+			} else { // Open the next file
```

Comment:
does this actually make a difference re correctness? or we're just saving ourselves from unnecessarily opening the next file sometimes?

Reply:
yes, this is more correct. opening of next file may result in an error (?). plus "saving ourselves from unnecessarily opening the next file sometimes", of course